pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -255,14 +256,19 @@ func (k *Kaniko) compileJobSpec(namespace string,
 	}
 
 	// Add build options args
-	for k, v := range buildOptions.BuildArgs {
-		buildArgs = append(buildArgs, fmt.Sprintf("--build-arg=%s=%s", k, v))
+	for buildArgName, buildArgValue := range buildOptions.BuildArgs {
+		buildArgs = append(buildArgs, fmt.Sprintf("--build-arg=%s=%s", buildArgName, buildArgValue))
 	}
 
 	tmpFolderVolumeMount := v1.VolumeMount{
 		Name:      "tmp",
 		MountPath: "/tmp",
 	}
+	kanikoWorkingDirFolderVolumeMount := v1.VolumeMount{
+		Name:      "kaniko-wd",
+		MountPath: "/kaniko",
+	}
+	buildArgs = append(buildArgs, fmt.Sprintf("--kaniko-dir=%s", kanikoWorkingDirFolderVolumeMount.MountPath))
```

Comment:
I think this also needs to be added to the Volume Mounts in the `kanikoJobSpec` below.

Also - Does this create the path if it doesn't exist?
If so, with what permissions?

Reply:
This directory already exists with 777, until security context is truly fix on kaniko's project - i will set that hard coded to /kaniko to avoid confusion + add comment explaining the 777

Reply:
actually removed. by default the working dir is `kaniko`. no need for this explicitly.